app/controllers/spotlight/concerns/catalog_search_context.rb

```diff
@@ -35,24 +33,21 @@ def current_search_session_from_browse_category?
           current_search_session.query_params['id']
       end
 
-      def current_search_session_from_page?
-        current_search_session &&
-          current_search_session.query_params['action'] == 'show' &&
-          current_search_session.query_params['controller'].ends_with?('_pages')
+      def current_page_from_page?
+        page_referrer &&
+          page_referrer[:action] == 'show' &&
+          page_referrer[:controller].ends_with?('_pages')
       end
 
-      def current_search_session_from_home_page?
```

Comment:
I couldn't figure out a reason why we need this. We guard against this case elsewhere already.